### DISCUSSION TITLE
如何在 JavaScript 中优雅地创建并初始化数组

### DISCUSSION THREAD

**Discussion by giscus** [darkyzhou/blog-comments#3] [General]
# 如何在 JavaScript 中优雅地创建并初始化数组

你知道在 JavaScript 中创建并初始化数组有多少种写法吗？我不是孔乙己，但也许真的有一种「最理想」的写法。

https://darkyzhou.net/articles/js-array-creation

**Comment by jacob-lcs** [comment]
开头的孔乙己写的太好了😂

**Comment by B41dr** [comment]
个人认为整篇文章在性能方面的理解存在问题。

1.  JS 使用 node 进行 run 时，创建数组的流程及其性能表现确实取决于 V8 的的优化手段，但是 V8 依赖于容器的架构，X86_64、 ARM、 RISC-V 等架构下 V8 的执行机制又不一样，不同版本的同种架构可能还会存在不可消解的变更，又导致了 V8 入口侧版本号等入口 ID 的 polyfill。“「带洞数组的性能可能劣于紧密数组」的事实” 这句话是病句，这件事情最量化的手段是用生产大集群或者私有服务器进行表现验证，我使用 v8: '11.3.244.8-node.23'、node: '20.15.1' 、ARM_M1Pro_16G 进行了验证，似乎不太和博主的结论一致。

2. 整篇文章措辞优秀。是一篇好的分享，学习了。
